packages/flutter/lib/src/widgets/implicit_animations.dart

```diff
@@ -423,6 +442,20 @@ abstract class ImplicitlyAnimatedWidgetState<T extends ImplicitlyAnimatedWidget>
   ///
   /// Any properties that depend upon tweens created by [forEachTween] should be
   /// updated within [didUpdateTweens], not within [forEachTween].
+  ///
+  /// This method will be called both:
+  ///
+  ///  1. After the tweens are _initially_ constructed (by
+  ///     the `constructor` argument to the [TweenVisitor] that's passed to
+  ///     [forEachTween]). In this case, the tweens are likely to contain only
+  ///     a [Tween.begin] value and not a [Tween.end].
+  ///
+  ///  2. When the state's [widget] is updated, and one or more of the tweens
```

Comment:
This is the confusingly simple bit. The animation's don't start immediately. Each time the didUpdateWidget method of the ImplicitlyAnimatedWidgetState subclass runs, they might start, per the condition you've specified. In other words, each time the ImplicitlyAnimatedWidget subclass is rebuilt, the animations might start.

Reply:
Yep.  I tried to add a lot more documentation to `forEachTween()` to make this clear.